src/functions/Mock.ps1

```diff
@@ -452,43 +468,68 @@ function Should-InvokeInternal {
+                    Actual   = "${commandName}${moduleMessage} was called $($matchingCalls.count) times"
+                    Because  = Format-Because $Because
+                }
             }
         }
     }
     else {
         if ($matchingCalls.Count -ne $Times -and ($Exactly -or ($Times -eq 0))) {
-            return [PSCustomObject] @{
+            return [Pester.ShouldResult] @{
                 Succeeded      = $false
                 FailureMessage = "Expected ${commandName}${moduleMessage} to be called $Times times exactly,$(Format-Because $Because) but was called $($matchingCalls.Count) times"
+                ExpectResult   = [Pester.ShouldExpectResult]@{
+                    Expected = "${commandName}${moduleMessage} to be called $Times times exactly"
+                    Actual   = "${commandName}${moduleMessage} was called $($matchingCalls.count) times"
+                    Because  = Format-Because $Because
+                }
             }
         }
         elseif ($matchingCalls.Count -lt $Times) {
-            return [PSCustomObject] @{
+            return [Pester.ShouldResult] @{
                 Succeeded      = $false
                 FailureMessage = "Expected ${commandName}${moduleMessage} to be called at least $Times times,$(Format-Because $Because) but was called $($matchingCalls.Count) times"
+                ExpectResult   = [Pester.ShouldExpectResult]@{
+                    Expected = "${commandName}${moduleMessage} to be called at least $Times times"
+                    Actual   = "${commandName}${moduleMessage} was called $($matchingCalls.count) times"
+                    Because  = Format-Because $Because
+                }
             }
         }
         elseif ($filterIsExclusive -and $nonMatchingCalls.Count -gt 0) {
-            return [PSCustomObject] @{
+            return [Pester.ShouldResult] @{
                 Succeeded      = $false
                 FailureMessage = "Expected ${commandName}${moduleMessage} to only be called with with parameters matching the specified filter,$(Format-Because $Because) but $($nonMatchingCalls.Count) non-matching calls were made"
+                ExpectResult   = [Pester.ShouldExpectResult]@{
+                    Expected = "${commandName}${moduleMessage} to only be called with with parameters matching the specified filter"
+                    Actual   = "${commandName}${moduleMessage} was called $($nonMatchingCalls.Count) times with non-matching parameters"
+                    Because  = Format-Because $Because
+                }
             }
         }
     }
 
-    return [PSCustomObject] @{
+    return [Pester.ShouldResult] @{
         Succeeded      = $true
-        FailureMessage = $null
+        FailureMessage = $null #TODO Pester 6: This should be ommitted for consistency but someone may have a strict check for null that may fail on the absence of this property.
```

Comment:
Should be okay to remove. This is internal function. Output-object not surfaced to user without calling `Should-Invoke/InvokeInternal` inside Pester-module which is unsupported

Reply:
757b66043cce99e25eed591a52d7c46376000d57 done